plugins/global-search/src/utils/indexer/IndexerProvider.tsx

```diff
@@ -0,0 +1,72 @@
+import { startTransition, useEffect, useMemo, useState } from "react"
+import { IndexerContext } from "./context"
+import type { IndexerEvents } from "./indexer"
+import { GlobalSearchIndexer } from "./indexer"
+import type { IndexEntry } from "./types"
+
+/**
+ * Creates an indexer instance and provides it to the children.
+ * Manages the index in a React state, so that it can use reactivity to show the results
+ */
+export function IndexerProvider({ children }: { children: React.ReactNode }) {
+    const indexer = useMemo(() => new GlobalSearchIndexer(), [])
```

Comment:
nit: should this be `useState()` too?

> However, since useMemo is performance optimization, not a semantic guarantee, React may throw away the cached value if [there is a specific reason to do that](https://react.dev/reference/react/useMemo#caveats).

Reply:
It can even be a `useRef` here, there is no tracking needed.